### PR TITLE
[14.0] product_import: Fix product's company

### PR DESCRIPTION
**Pull request by florentx**
And also:

- [ ] OCA/edi#768

### CODE REVIEW

**Review by mmequignon — APPROVED** [review]

**Review by TDu — APPROVED** [review]

File: product_import/wizard/product_import.py [hunk 1]
```diff
@@ -147,9 +147,12 @@ def _prepare_supplierinfo(self, seller_info, product):
 
     @api.model
     def _prepare_product(self, parsed_product, chatter_msg, seller=None):
+        product_company = self.env.context.get("product_company", False)
```
Comment (grindtildeath) [inline product_import/wizard/product_import.py]: I find it weird to default to a False value and then pass it to `with_company` that won't actually change the company if we have a False value. Wouldn't it make more sense to default to `self.env.company`?

**Review by grindtildeath — APPROVED** [review]
Not blocking

File: product_import/wizard/product_import.py (hunk 1 above)
Comment (florentx) [reply product_import/wizard/product_import.py]: I've refactored, and no longer use `with_company` here.

File: product_import/wizard/product_import.py [hunk 2]
```diff
@@ -215,7 +231,7 @@ def import_button(self):
             raise UserError(_("This catalogue doesn't have any product!"))
         company_id = self._get_company_id(catalogue)
         seller = self._get_seller(catalogue)
-        self.with_context(company_id=company_id)._create_products(
+        self.with_context(product_company=company_id)._create_products(
```
Comment (simahawk) [inline product_import/wizard/product_import.py]:
```suggestion
        self.with_context(product_company_id=company_id)._create_products(
```
to be more consistent

**Pull request merged**